src/components/material.js

```diff
@@ -152,11 +152,26 @@ module.exports.Component = registerComponent('material', {
    * @returns {object} Material.
    */
   setMaterial: function (material) {
-    var mesh = this.el.getOrCreateObject3D('mesh', THREE.Mesh);
+    var el = this.el;
+    var mesh;
     var system = this.system;
+
     if (this.material) { disposeMaterial(this.material, system); }
-    this.material = mesh.material = material;
+
+    this.material = material;
     system.registerMaterial(material);
+
+    // Set on mesh. If mesh does not exist, wait for it.
+    mesh = el.getObject3D('mesh');
+    if (mesh) {
+      mesh.material = material;
+    } else {
+      el.addEventListener('object3dset', function waitForMesh (evt) {
+        if (evt.detail.type !== 'mesh') { return; }
+        el.getObject3D('mesh').material = material;
+        el.removeEventListener('object3dset', waitForMesh);
+      });
```

Comment:
According to http://caniuse.com/#feat=once-event-listener, support for the `addEventListener(type, fn, {once: true})` option is good enough to start using.  MDN claims Edge doesn't have it yet, so we should check on that first.

Reply:
Too bad it wouldn't work since we have to do the type and target checks.

Reply:
Ah, yeah. 🙁